### PR TITLE
Decorator `Modal`

### LINKED ISSUE #17993: Decorator: Modal

### PR DESCRIPTION
**Pull request by guidari**
Closes #17993 

Added decorator to `Modal` 

#### Changelog

**New**

- New `decorator` prop added to `Modal` component

**Changed**

- Deprecated `slug` prop
- Update rendering slug to render the decorator component but still set the size if it's an AILabel
- Updated test file
- Added an `block-size` to the style to fix the alignment

#### Testing / Reviewing

- Make sure the `Modal` still work as expected and the one with `decorator` also works fine

- [x] Delete `Modal` with `Tooltip`
- [x] Delete `Datepicker` with `Tooltip`

### CODE REVIEW

File: packages/styles/scss/components/modal/_modal.scss [hunk 1]
```diff
@@ -366,31 +364,33 @@
   }
 
   // Slug styles
-  .#{$prefix}--modal--slug.#{$prefix}--modal {
+  .#{$prefix}--modal--slug,
+  .#{$prefix}--modal--decorator:has(.#{$prefix}--ai-label).#{$prefix}--modal {
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  .#{$prefix}--modal--slug.#{$prefix}--modal,
  .#{$prefix}--modal--decorator:has(.#{$prefix}--ai-label).#{$prefix}--modal {
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 2]
```diff
@@ -366,31 +364,33 @@
...
     background-color: $ai-overlay;
   }
 
-  .#{$prefix}--modal--slug .#{$prefix}--modal-container {
+  .#{$prefix}--modal--slug,
+  .#{$prefix}--modal--decorator .#{$prefix}--modal-container {
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  .#{$prefix}--modal--slug .#{$prefix}--modal-container ,
  .#{$prefix}--modal--decorator:has(.#{$prefix}--ai-label) .#{$prefix}--modal-container {
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 3]
```diff
@@ -366,31 +364,33 @@
...
     background-color: $layer;
-    box-shadow:
-      inset 0 -80px 70px -65px $ai-inner-shadow,
+    box-shadow: inset 0 -80px 70px -65px $ai-inner-shadow,
       0 24px 40px -24px $ai-drop-shadow;
   }
 
-  .#{$prefix}--modal--slug
+  .#{$prefix}--modal--slug,
+  .#{$prefix}--modal--decorator
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  .#{$prefix}--modal--slug .#{$prefix}--modal-container:has(.#{$prefix}--modal-footer),
  .#{$prefix}--modal--decorator:has(.#{$prefix}--ai-label)
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 4]
```diff
@@ -366,31 +364,33 @@
...
 
-    box-shadow:
-      inset 0 -80px 0 -16px $layer,
+    box-shadow: inset 0 -80px 0 -16px $layer,
       inset 0 -160px 70px -65px $ai-inner-shadow,
       0 24px 40px -24px $ai-drop-shadow;
   }
 
-  .#{$prefix}--modal--slug
+  .#{$prefix}--modal--slug,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  .#{$prefix}--modal--slug .#{$prefix}--modal-content.#{$prefix}--modal-scroll-content,
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 5]
```diff
@@ -424,6 +429,10 @@
     inset-inline-end: convert.to-rem(48px);
   }
 
+  .#{$prefix}--modal-inner-wrapper--decorator:not(:has(.#{$prefix}--ai-label)) {
+    inset-block-start: 1rem;
+  }
+
   .#{$prefix}--modal--slug
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
.#{$prefix}--modal--decorator
    .#{$prefix}--modal-content--overflow-indicator::before,
  .#{$prefix}--modal--decorator .#{$prefix}--modal-content--overflow-indicator {
  .#{$prefix}--modal--slug
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 6]
```diff
@@ -405,12 +405,17 @@
 
   .#{$prefix}--modal-header
     > .#{$prefix}--ai-label:has(+ .#{$prefix}--modal-close-button),
+  .#{$prefix}--modal-inner-wrapper--decorator:has(
+      + .#{$prefix}--modal-close-button
+    ),
   .#{$prefix}--modal-header
     > .#{$prefix}--modal-close-button
     ~ .#{$prefix}--ai-label,
+  .#{$prefix}--modal-inner-wrapper--decorator,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
.#{$prefix}--modal-header
    > .#{$prefix}--modal-close-button
    ~ .#{$prefix}--modal-inner-wrapper--decorator > *,
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 7]
```diff
@@ -405,12 +405,17 @@
...
   .#{$prefix}--modal--slug
     .#{$prefix}--modal-container-body
     > .#{$prefix}--ai-label,
+  .#{$prefix}--modal-inner-wrapper--decorator,
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
.#{$prefix}--modal--decorator
    .#{$prefix}--modal-container-body
    > .#{$prefix}--modal-inner-wrapper--decorator > *
```

File: packages/styles/scss/components/modal/_modal.scss [hunk 8]
```diff
@@ -405,12 +405,17 @@
 
   .#{$prefix}--modal-header
     > .#{$prefix}--ai-label:has(+ .#{$prefix}--modal-close-button),
+  .#{$prefix}--modal-inner-wrapper--decorator:has(
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  .#{$prefix}--modal-header
      >
  .#{$prefix}--modal-inner-wrapper--decorator:has(
```

**Review by ariellalgilmore — COMMENTED** [review]
Needs some style cleanup... lmk if you want to go over my comments! Double check that the slug property and decorator prop show the same results even though the slug is deprecated now

File: packages/styles/scss/components/modal/_modal.scss [hunk 9]
```diff
@@ -365,18 +365,14 @@
     margin: 0;
   }
 
-  // AIlabel styles
+  // Slug styles
```
Comment (ariellalgilmore) [inline packages/styles/scss/components/modal/_modal.scss]:
```suggestion
  // AIlabel styles
```

**Review by ariellalgilmore — APPROVED** [review]
Looks great! just need to remove the tooltip modal button and remove the revert on AILabel before merging

**Review by annawen1 — APPROVED** [review]

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**